### PR TITLE
7989 - Returning AnalysisResult or DataArtifact instead of BlackboardArtifact

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 1]
```diff
@@ -3668,42 +3677,48 @@ public long getBlackboardArtifactsTypeCount(int artifactTypeID, long dataSourceI
 	 *                          within tsk core and artifacts could not be
 	 *                          queried
 	 */
+	@Deprecated
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:3680]: We should at a @ deprecated item to the doc as well, and it should refer developers to the method that they should use instead. This comment applies to all of the methods given @ Deprecated annotations by this PR.

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 2]
```diff
@@ -3488,7 +3489,9 @@ public DataSource getDataSource(long objectId) throws TskDataException, TskCoreE
 	 * @throws TskCoreException
 	 */
 	public ArrayList<BlackboardArtifact> getBlackboardArtifacts(int artifactTypeID) throws TskCoreException {
-		return getArtifactsHelper("blackboard_artifacts.artifact_type_id = " + artifactTypeID);
+		ArrayList<BlackboardArtifact> artifacts = new ArrayList<>();
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:3492]: Can this method be deprecated with a referral to an equivalent Blackboard method?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 3]
```diff
@@ -6178,27 +6168,26 @@ public BlackboardArtifact getArtifactById(long id) throws TskCoreException {
 	 *                          core and file could not be queried
 	 */
 	public BlackboardArtifact getArtifactByArtifactId(long id) throws TskCoreException {
-		CaseDbConnection connection = null;
-		ResultSet rs = null;
+		String query = "SELECT artifact_type_id, artifact_obj_id WHERE artifact_id = " + id;
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:6171]: Can this method be deprecated with a referral to an equivalent Blackboard method?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 4]
```diff
@@ -4326,38 +4319,64 @@ public long getBlackboardArtifactsCount(ARTIFACT_TYPE artifactType, long obj_id)
 	 *                          within TSK core
 	 */
 	public List<BlackboardArtifact> getBlackboardArtifacts(ARTIFACT_TYPE artifactType, BlackboardAttribute.ATTRIBUTE_TYPE attrType, String value) throws TskCoreException {
-		CaseDbConnection connection = null;
-		Statement s = null;
-		ResultSet rs = null;
+
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:4322]: Can this method be deprecated?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 5]
```diff
@@ -4374,40 +4393,17 @@ public long getBlackboardArtifactsCount(ARTIFACT_TYPE artifactType, long obj_id)
...
-				 */
-				throw new TskCoreException("No blackboard artifact with id " + artifactID);
-			}
-		} catch (SQLException ex) {
-			throw new TskCoreException("Error getting a blackboard artifact. " + ex.getMessage(), ex);
-		} finally {
-			closeResultSet(rs);
-			closeConnection(connection);
-			releaseSingleUserCaseReadLock();
+		List<DataArtifact> dataArtifacts = blackboard.getDataArtifactsWhere("artifacts.artifact_id = " + artifactID);
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:4396]: Can this method be deprecated with a referral to an equivalent Blackboard method?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 6]
```diff
@@ -5163,35 +5159,37 @@ String addSourceToArtifactAttribute(BlackboardAttribute attr, String source) thr
 	 *                          within tsk core \ref query_database_page
 	 */
 	public ArrayList<BlackboardArtifact> getMatchingArtifacts(String whereClause) throws TskCoreException {
-		CaseDbConnection connection = null;
-		Statement s = null;
-		ResultSet rs = null;
+		String query = "SELECT blackboard_artifacts.artifact_id AS artifact_id, "
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5162]: Can this method be deprecated with a referral to an equivalent Blackboard method?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 7]
```diff
@@ -12678,6 +12632,46 @@ String getInsertOrIgnoreSQL(String sql) {
 				throw new UnsupportedOperationException("Unsupported DB type: " + getDatabaseType().name());
 		}
 	}
+	
+	/**
+	 * This helper method that returns a list of Blackboard artifact. The method
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:12637]: The grammar is a little off in the first sentence here.

**Review by rcordovano — CHANGES_REQUESTED** [review]

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java (hunk 6 above)
Comment (kellykelly3) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5162]: @rcordovano This one is tricky to say if we can deprecate. I am going to say maybe.

In the blackboard methods, the "Where" methods use an alias for the table names.  This method does not use tables alias, it uses the direct table name.  Therefore someone calling this method would pass in a where that refers to the table name i.e. blackboard_aritifacts.obj_id whereas the methods in blackboard expect the user to know the alias that is being used by those methods.

Comment (rcordovano) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5175]: Let's stay away from tricky right now. We will revisit this later in the context of the story about refactoring SleuthkitCase/Blackboard.

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 8]
```diff
@@ -3615,38 +3620,44 @@ public long getBlackboardArtifactsTypeCount(int artifactTypeID, long dataSourceI
 	 *                          within tsk core and artifacts could not be
 	 *                          queried
 	 */
+	@Deprecated
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:3623]: Missing the doc here...

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 9]
```diff
@@ -5316,21 +5327,15 @@ PreparedStatement createInsertArtifactStatement(int artifact_type_id, long obj_i
 
 	@Deprecated
 	BlackboardArtifact newBlackboardArtifact(int artifact_type_id, long obj_id, String artifactTypeName, String artifactDisplayName, long data_source_obj_id, CaseDbConnection connection) throws TskCoreException {
-		acquireSingleUserCaseWriteLock();
+		BlackboardArtifact.Type type = getArtifactType(artifact_type_id);
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5330]: This is not public API, do we even need this or can it just be deleted?

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 10]
```diff
@@ -8512,22 +8521,22 @@ public LayoutFile addLayoutFile(String fileName,
...
-	 * 
+	 * Given a Content object, return its data source object ID. For
+	 * AbstractFiles, this simply returns the data source ID field.
+	 *
 	 * @param connection A case database connection.
 	 * @param content    The content to look up the data source object ID.
-	 * 
-	 * @return 
+	 *
+	 * @return
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:8532]: The return type is undocumented.

**Review by rcordovano — CHANGES_REQUESTED** [review]

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java (hunk 9 above)
Comment (kellykelly3) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5330]: @rcordovano This method is used by other versions of newBlackboardArtifact. It seems to be a helper method, I have added documentation and made the method private.

**Review by gdicristofaro — APPROVED** [review]

**Pull request merged**